azurelinuxagent/ga/update.py

```diff
@@ -586,6 +578,14 @@ def _get_pid_files(self):
         pid_files.sort(key=lambda f: int(pid_re.match(os.path.basename(f)).group(1)))
         return pid_files
 
+    @property
+    def is_running(self):
```

Comment:
The unit tests were controlling the execution of the main loop by mocking sleep (which is called at the end of the main loop). That was a global mock, which made it extremely fragile. This simple refactoring broke over 50 tests.

As an alternative, I exposed the variable that controls the main loop as a property and now the tests mock that property instead of mocking sleep.